src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs

```diff
@@ -53,9 +54,26 @@ public static void AddCastleWindsor(this IServiceCollection services, IWindsorCo
 		/// <param name="container">Windsor container to register framework types in</param>
 		public static void UseCastleWindsor<TTypeAssembly>(this IApplicationBuilder app, IWindsorContainer container)
 		{
-			container.Register(Classes.FromAssemblyInThisApplication(typeof(TTypeAssembly).Assembly).BasedOn<Controller>().LifestyleScoped());
-			container.Register(Classes.FromAssemblyInThisApplication(typeof(TTypeAssembly).Assembly).BasedOn<ViewComponent>().LifestyleTransient());
-			container.Register(Classes.FromAssemblyInThisApplication(typeof(TTypeAssembly).Assembly).BasedOn<TagHelper>().LifestyleTransient());
+			var assembly = typeof(TTypeAssembly).Assembly;
+			UseCastleWindsor(container, assembly);
+		}
+
+		/// <summary>
+		/// Use this to register all Controllers, ViewComponents and TagHelpers. Will register the entry assembly and its referenced items
+		/// </summary>
+		/// <param name="app">Application builder retained as extension</param>
+		/// <param name="container">Windsor container to register framework types in</param>
+		public static void UseCastleWindsor(this IApplicationBuilder app, IWindsorContainer container)
+		{
+			var assembly = Assembly.GetEntryAssembly();
+			UseCastleWindsor(container, assembly);
+		}
+
+		private static void UseCastleWindsor(IWindsorContainer container, Assembly assembly)
+		{
+			container.Register(Classes.FromAssemblyInThisApplication(assembly).BasedOn<Controller>().LifestyleScoped());
+			container.Register(Classes.FromAssemblyInThisApplication(assembly).BasedOn<ViewComponent>().LifestyleTransient());
+			container.Register(Classes.FromAssemblyInThisApplication(assembly).BasedOn<TagHelper>().LifestyleTransient());
 		}
 
```

Comment:
^^ This method should probably be moved to the bottom of the file because it is private. 

https://github.com/castleproject/Home/blob/master/coding-standards.md#type-members